provider/image.go

```diff
@@ -175,15 +175,22 @@ func (p *dockerNativeProvider) dockerBuild(ctx context.Context,
 	if err != nil {
 		return "", nil, err
 	}
-	// Quick and dirty auth; we can also preconfigure the client itself I believe
-	// TODO: use auth pattern as above to use default auth
-	var authConfig = types.AuthConfig{
-		Username:      img.Registry.Username,
-		Password:      img.Registry.Password,
-		ServerAddress: img.Registry.Server,
+
+	// authentication for registry push
+	// we check if the user set creds in the Pulumi program, and use those preferentially
+	var pushAuthConfig types.AuthConfig
+
+	if img.Registry.Username != "" && img.Registry.Password != "" {
+		pushAuthConfig.Username = img.Registry.Username
+		pushAuthConfig.Password = img.Registry.Password
+		pushAuthConfig.ServerAddress = img.Registry.Server
+	} else {
```

Comment:
either is valid - neither field is required per the schema. If either field is empty, we default to the host credentials. 🤔 warning might make sense because using host credentials isn't as obvious. 👍

Reply:
"valid" was perhaps the wrong word to use. What I should have said, and what I meant, was "does it ever make sense for the user to set a username without a password or vice versa."

My goal is to make sure that if the user does something that doesn't make sense, we will let them know.